bootstrap fails silently

With #217 being merged, `lerna bootstrap` now silently fails without giving a reason.
```
➜  npmE lerna bootstrap
Lerna v2.0.0-beta.22
Independent Versioning Mode
Linking all dependencies
Errored while running BootstrapCommand.execute
1
```
instead of:
```
Lerna v2.0.0-beta.22
Independent Versioning Mode
Linking all dependencies
normalizeTree             ▄ ╢░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░╟╟
npm ERR! Darwin 15.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "xxx@^0.0.0" "skatejs@1.0.0-beta.2"
npm ERR! node v6.2.1
npm ERR! npm  v3.9.3
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/xxx
```

with
```javascript
//  stdio: "ignore"
```
in https://github.com/lerna/lerna/pull/217/commits/86bb31e46d1012f82b76043350db4d83f7701844#diff-5463599b8d45d087bae97f0d5df54fdaR15

--- a/src/ChildProcessUtilities.js
+++ b/src/ChildProcessUtilities.js
@@ -33,8 +33,21 @@ export default class ChildProcessUtilities {
   }
 
   static spawn(command, args, opts, callback) {
-    child.spawn(command, args, objectAssign({
+    let stderr = "";
+
+    const childProcess = child.spawn(command, args, objectAssign({
       stdio: "inherit"
-    }, opts)).on("close", callback).on("error", () => null);
+    }, opts))
+      .on("error", () => {})
+      .on("exit", code => {
+        callback(code && (stderr || `Command failed: ${command} ${args.join(" ")}`))
+      });
+
+    // By default stderr is inherited from us (just sent to _our_ output).
+    // If the caller overrode that to "pipe", then we'll gather that up and
+    // call back with it in case of failure.
+    if (childProcess.stderr) {
+      childProcess.stderr.setEncoding("utf8").on("data", chunk => stderr += chunk);
+    }
   }
 }

--- a/src/NpmUtilities.js
+++ b/src/NpmUtilities.js
@@ -12,7 +12,7 @@ export default class NpmUtilities {
 
     const opts = {
       cwd: directory,
-      stdio: "ignore"
+      stdio: ["ignore", "ignore", "pipe"],
     }
 
     ChildProcessUtilities.spawn("npm", args, opts, callback);

--- a/test/BootstrapCommand.js
+++ b/test/BootstrapCommand.js
@@ -11,6 +11,8 @@ import stub from "./_stub";
 
 import assertStubbedCalls from "./_assertStubbedCalls";
 
+const STDIO_OPT = ["ignore", "ignore", "pipe"];
+
 describe("BootstrapCommand", () => {
 
   describe("dependencies between packages in the repo", () => {
@@ -28,7 +30,7 @@ describe("BootstrapCommand", () => {
 
       assertStubbedCalls([
         [ChildProcessUtilities, "spawn", { nodeCallback: true }, [
-          { args: ["npm", ["install", "package-1@^0.0.0"], { cwd: path.join(testDir, "packages/package-4"), stdio: "ignore" }] }
+          { args: ["npm", ["install", "package-1@^0.0.0"], { cwd: path.join(testDir, "packages/package-4"), stdio: STDIO_OPT }] }
         ]]
       ]);
 
@@ -77,7 +79,7 @@ describe("BootstrapCommand", () => {
 
       assertStubbedCalls([
         [ChildProcessUtilities, "spawn", { nodeCallback: true }, [
-          { args: ["npm", ["install", "package-1@^0.0.0"], { cwd: path.join(testDir, "packages/package-4"), stdio: "ignore" }] }
+          { args: ["npm", ["install", "package-1@^0.0.0"], { cwd: path.join(testDir, "packages/package-4"), stdio: STDIO_OPT }] }
         ]]
       ]);
 
@@ -132,7 +134,7 @@ describe("BootstrapCommand", () => {
       let installed = false;
       stub(ChildProcessUtilities, "spawn", (command, args, options, callback) => {
         assert.deepEqual(args, ["install", "external@^1.0.0"])
-        assert.deepEqual(options, { cwd: path.join(testDir, "packages/package-1"), stdio: "ignore" })
+        assert.deepEqual(options, { cwd: path.join(testDir, "packages/package-1"), stdio: STDIO_OPT })
         installed = true;
         callback();
       });

